src/html-pipe.js

```diff
@@ -54,6 +54,31 @@ async function fetchContentWith404Fallback(state, req, res) {
   }
 }
 
+/**
+ * Loads the resource from the content bus but only handles the redirect; otherwise applies
+ * a 404 fallback.
+ * @param state
+ * @param req
+ * @param res
+ * @returns {Promise<void>}
+ */
+async function fetchContentRedirectWith404Fallback(state, req, res) {
+  // force load from content bus again
+  state.content.sourceBus = 'content';
+  const prevError = res.error;
+  try {
+    await fetchContent(state, req, res);
+  } finally {
+    state.content.sourceBus = 'code';
+  }
+  if (res.status !== 301) {
+    // force 404
+    res.status = 404;
+    res.error = prevError;
```

Comment:
this is the error it received initially when trying to load the resource from the code bus